From d24dc9bebaefbd8abf32707bad5efc0c811c0cd4 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 18 Jan 2023 13:04:56 -0800
Subject: [PATCH] Fix -Wincompatible-function-pointer-types

Fixes warnings with clang16
testfile.c:576:31: error: incompatible function pointer types passing 'PRInt32 (void *)' (aka 'int (void *)') to parameter of type 'void (*)(void *)' [-Wincompatible-function-pointe r-types]

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 pr/tests/testfile.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pr/tests/testfile.c b/pr/tests/testfile.c
index 79d8a05..42aa6a7 100644
--- a/pr/tests/testfile.c
+++ b/pr/tests/testfile.c
@@ -123,7 +123,7 @@ PRThread* create_new_thread(PRThreadType type,
     if (native_thread) {
 #if defined(_PR_PTHREADS)
         pthread_t tid;
-        if (!pthread_create(&tid, NULL, start, arg)) {
+        if (!pthread_create(&tid, NULL, (void*(*)(void*))start, arg)) {
             return((PRThread *) tid);
         }
         else {
@@ -573,7 +573,7 @@ static PRInt32 RunDirTest(void)
 
         thrarg.done= 0;
         t = create_new_thread(PR_USER_THREAD,
-                              DirTest, &thrarg,
+                              (void (*)(void *))DirTest, &thrarg,
                               PR_PRIORITY_NORMAL,
                               PR_LOCAL_THREAD,
                               PR_UNJOINABLE_THREAD,
-- 
2.39.1

